back-nest/src/sсrapers/scraper-rozetka.service.ts

```diff
@@ -77,28 +77,28 @@ export class ScraperRozetkaService {
     }
   }
 
-  async clearRozetkaProducts(): Promise<void> {
-    try {
-      this.logger.log('Deleting products with source "ROZETKA"');
-      await this.productRepository.delete({ source: 'ROZETKA' });
-      this.logger.log('Successfully deleted products with source "ROZETKA"');
-    } catch (error) {
-      this.logger.error(
-        'Failed to delete products with source "ROZETKA"',
-        error,
-      );
-    }
-  }
-
-  async saveProductsToDB(products: Product[], source: string): Promise<void> {
-    if (!products.length) {
+  async saveProductsToDB(products: Product[]): Promise<void> {
+    if (!products || products.length === 0) {
+      this.logger.log('No products to save.');
       return;
     }
 
-    await this.productRepository.save(products);
-    this.logger.log(`Saved ${products.length} products from ${source} to DB.`);
+    const operations = products.map(async (product) => {
+      const existingProduct = await this.productRepository.findOne({
+        where: { title: product.title, source: product.source },
+      });
+
+      if (existingProduct) {
+        await this.productRepository.update(existingProduct.id, product);
+        this.logger.log(`Updated product with title: ${product.title}`);
+      } else {
+        await this.productRepository.save(product);
+        this.logger.log(`Created new product with title: ${product.title}`);
+      }
+    });
```

Comment:
Use inversion for that type of conditions to avoid nested code as much as possible

```suggestion
      if (!existingProduct) {
        await this.productRepository.save(product);
        this.logger.log(`Created new product with title: ${product.title}`);
        return
      }
        await this.productRepository.update(existingProduct.id, product);
        this.logger.log(`Updated product with title: ${product.title}`);
    });
```